certification/internal/bundle/bundle.go

```diff
@@ -198,16 +198,66 @@ func GetSupportedInstallModes(ctx context.Context, csvReader io.Reader) (map[str
 	return installedModes, nil
 }
 
+func ExtractImagesFromBundle(ctx context.Context, csvReader io.Reader) ([]string, error) {
+	csvBytes, err := io.ReadAll(csvReader)
+	if err != nil {
+		return nil, fmt.Errorf("could not read CSV from path: %v", err)
+	}
+
+	var csv ClusterServiceVersion
+	if err := yaml.Unmarshal(csvBytes, &csv); err != nil {
+		return nil, fmt.Errorf("could not unmarshal CSV: %v", err)
+	}
+
+	imageMap := make(map[string]struct{}, len(csv.Spec.RelatedImages))
+	// This will trim down any duplicates
+	for _, v := range csv.Spec.RelatedImages {
+		imageMap[v.Image] = struct{}{}
+	}
+	for _, deployment := range csv.Spec.Install.Spec.Deployments {
+		for _, container := range deployment.Spec.Template.Spec.Containers {
+			imageMap[container.Image] = struct{}{}
+		}
+	}
+	images := make([]string, 0, len(imageMap))
+	for k := range imageMap {
+		images = append(images, k)
+	}
+	return images, nil
+}
+
 type ClusterServiceVersion struct {
 	Spec ClusterServiceVersionSpec `yaml:"spec"`
 }
 
 type ClusterServiceVersionSpec struct {
 	// InstallModes specify supported installation types
-	InstallModes []InstallMode `yaml:"installModes,omitempty"`
+	InstallModes  []InstallMode  `yaml:"installModes,omitempty"`
+	RelatedImages []RelatedImage `yaml:"relatedImages,omitempty"`
+	Install       Install        `yaml:"install,omitempty"`
 }
 
 type InstallMode struct {
 	Type      string `yaml:"type"`
 	Supported bool   `yaml:"supported"`
 }
+
+type RelatedImage struct {
+	Image string `yaml:"image"`
+}
+
+type Install struct {
+	Spec struct {
+		Deployments []struct {
+			Spec struct {
+				Template struct {
+					Spec struct {
+						Containers []struct {
+							Image string `yaml:"image,omitempty"`
+						} `yaml:"containers,omitempty"`
+					} `yaml:"spec,omitempty"`
+				} `yaml:"template,omitempty"`
+			} `yaml:"spec,omitempty"`
+		} `yaml:"deployments,omitempty"`
+	} `yaml:"spec,omitempty"`
+}
```

Comment:
At this point, why don't we just start using the upstream API definitions? I understand not using it when we only care about very specific fields, and fewer of the structs, but we're adding more and more here. 

https://github.com/operator-framework/api/blob/master/pkg/operators/v1alpha1/clusterserviceversion_types.go

Reply:
We're already pulling in that dep, so probably not a bad idea.